desc/optimize/_constraint_wrappers.py

```diff
@@ -800,8 +757,6 @@ def jac_scaled(self, x, constants=None):
         # Jacobian matrices wrt combined state vectors
         Fx = self._constraint.jac_scaled(xf, constants[1])
         Gx = self._objective.jac_scaled(xg, constants[0])
-        Fx = align_jacobian(Fx, self._constraint, self._objective)
-        Gx = align_jacobian(Gx, self._objective, self._constraint)
```

Comment:
I don't understand how we are ensuring these Jacobians are still aligned in the new code

Reply:
Both the objective and constraint are bound to the same things in `Optimizer.optimize`, and they always take the full set of optimizable params as args. The ordering of the parameters is deterministic, so they should always line up.